server/pkg/webapp/serve.go

```diff
@@ -148,7 +151,67 @@ func serveConfig(r chi.Router, config webappConfig) {
 	r.Get(options.CleanBase(config.appUrl, "custom.css"), func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Add("Content-Type", "text/css")
 
-		_, _ = fmt.Fprint(w, config.settings.UI.CustomCSS)
+		var stringsBuilder strings.Builder
+
+		assetFiles := assets.Files(logger.Default(), "")
+
+		// process BrandingSASS
+		if config.settings.UI.BrandingSASS != "" {
+			brandingSassString, err := jsonToSass(config.settings.UI.BrandingSASS)
+			if err != nil {
+				logger.Default().Error("failed to parse branding SASS", zap.Error(err))
+				return
+			}
+			stringsBuilder.WriteString(brandingSassString)
+		}
+
+		// Get SASS variables from the custom CSS editor and give them precedence over branding variables
+		variablesPattern := regexp.MustCompile(`(\$[a-zA-Z_-]+):\s*([^;]+);`)
+		customVariables := variablesPattern.FindAllString(config.settings.UI.CustomCSS, -1)
+
+		for _, customVariable := range customVariables {
+			stringsBuilder.WriteString(fmt.Sprintf("%s \n", customVariable))
+		}
+
+		// Boostrap, bootstrap-vue and custom variables scss files
+		fileNames := assets.FileNames("scss")
+		for _, fileName := range fileNames {
+			open, err := assetFiles.Open("scss/" + fileName)
+			if err != nil {
+				logger.Default().Error(fmt.Sprintf("failed to open asset %s file", fileName), zap.Error(err))
+				return
+			}
+			fileContent, err := io.ReadAll(open)
```

Comment:
I think you'll need to make a buffer over the file handle; IIRC go doesn't do that automagically and it greatly improves performance if you do it